### PR TITLE
Fix "more than one target found for cross-reference"

### LINKED ISSUE #211: Fix "more than one target found for cross-reference"
This causes lots of warnings during doc compilation. We must remove pyof version ambiguity in references. E.g.: `pyof.v0x01.common.header.Header` instead of `common.header.Header`. A relative reference would be better so the docstring can be reused in other pyof versions.

### PR DESCRIPTION
**Pull request by macartur**
FIX #211

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -374,10 +377,8 @@
 # Note: links to Python doc only work if you are online or have python.inv
 #     file. To download it, run:
 #     curl https://docs.python.org/3/objects.inv >python.inv
```
Comment (cemsbr) [inline docs/conf.py]: Comments after _if you are online_ doesn't make sense anymore.

File: pyof/foundation/base.py [hunk 2]
```diff
@@ -366,15 +367,16 @@ def replace_pyof_version(module_fullname, version):
...
-                module_fullname or if the module_fullname is not a 'OF version'
-                specific module.
-            new_module_fullname (str): The new module fullname, with the
-                replaced version, on the format "pyof.v0x01.common.header".
+            str  : module fullname
+                   The new module fullname, with the replaced version,
+                   on the format "pyof.v0x01.common.header". If the requested
+                   version is the same as the one of the module_fullname or if
+                   the module_fullname is not a 'OF version' specific module
+                   the return is None.
```
Comment (cemsbr) [inline pyof/foundation/base.py]: _...module, returns None._ Is it common practice to make a short first line and write more comments in a new line?

File: CHANGELOG.rst [hunk 3]
```diff
@@ -19,6 +19,8 @@ Removed
 
 Fixed
 =====
+Fixed some links using sphinx documentation.
+Fixed all sphinx warnings during builds.
```
Comment (cemsbr) [inline CHANGELOG.rst]: I would write something more high-level in the changelog like:
- Improved documentation: fixed links and build warnings.

File: pyof/v0x01/asynchronous/error_msg.py [hunk 4]
```diff
@@ -204,7 +203,12 @@ def pack(self, value=None):
             raise PackException(msg)
 
     def unpack(self, buff, offset=0):
-        """Unpack binary data into python object."""
+        """Unpack binary data into python object.
+
+        Parameters:
```
Comment (diraol) [inline pyof/v0x01/asynchronous/error_msg.py]: We are using `Args:` instead of `Parameters`?

File: pyof/v0x04/controller2switch/common.py [hunk 5]
```diff
@@ -358,7 +370,7 @@ def __init__(self, experimenter=None, exp_type=None):
 
         Args:
             experimenter: Experimenter ID which takes the same form as in
-                struct ofp_experimenter_header (class ExperimenterHeader).
+                struct ofp_experimenter_header (:class: `ExperimenterHeader`).
```
Comment (diraol) [inline pyof/v0x04/controller2switch/common.py]: Should this space after `:` exist?

File: pyof/v0x04/controller2switch/common.py [hunk 6]
```diff
@@ -457,9 +473,11 @@ def __init__(self, property_type=TableFeaturePropType.OFPTFPT_NEXT_TABLES,
         """Constructor of NextTablesProperty receives the parameters bellow.
 
         Args:
-            type(TableFeaturePropType): Property Type value of this instance.
-            next_table_ids(ListOfInstruction): List of InstructionGotoTable
-                                               instances.
+            type(~pyof.v0x04.controller2switch.common.TableFeaturePropType):
+                Property Type value of this instance.
+            next_table_ids
```
Comment (diraol) [inline pyof/v0x04/controller2switch/common.py]: Why haven't you used the backslash here, like on other cases you did used it?

File: pyof/v0x01/asynchronous/error_msg.py (hunk 4 above)
Comment (macartur) [inline pyof/v0x01/asynchronous/error_msg.py]: OK, I will change this.

File: pyof/v0x04/controller2switch/common.py (hunk 5 above)
Comment (macartur) [inline pyof/v0x04/controller2switch/common.py]: Sorry, my mistake.

File: pyof/v0x04/controller2switch/common.py (hunk 6 above)
Comment (macartur) [inline pyof/v0x04/controller2switch/common.py]: It's not necessary.I will fix this.

**Comment by diraol** [comment]
@macartur please, remember to use better names at the Pull Request, as much as a goog PR description.
Just using the related issue number does not help much.

**Comment by beraldoleal** [comment]
@macartur waiting for changes...

File: pyof/foundation/base.py [hunk 7]
```diff
@@ -49,9 +50,9 @@ def __init__(self, value=None, enum_ref=None):
         """The constructor takes the optional parameters below.
 
         Args:
-            value: The type's value.
-            enum_ref (:class:`type`): If :attr:`value` is from an Enum, specify
-                its type.
+            value : The type's value.
```
Comment (diraol) [inline pyof/foundation/base.py]: extra space here.

File: pyof/foundation/base.py [hunk 8]
```diff
@@ -49,9 +50,9 @@ def __init__(self, value=None, enum_ref=None):
...
+            enum_ref (type): If :attr:`value` is from an Enum, specify
+                             its type.
```
Comment (diraol) [inline pyof/foundation/base.py]: Is this the right indentation? Or it should be just a 4 spaces indentation?

File: pyof/foundation/base.py [hunk 9]
```diff
@@ -346,8 +347,8 @@ def get_pyof_version(module_fullname):
                 (e.g.: pyof.v0x01.common.header)
 
         Returns:
-            version (str): The module version, on the format 'v0x0?' if any. Or
-            None (None): If there isn't a version on the fullname.
+            str : The module version, on the format 'v0x0?' if any. Or None
```
Comment (diraol) [inline pyof/foundation/base.py]: Extra space

File: pyof/foundation/base.py [hunk 10]
```diff
@@ -366,15 +367,16 @@ def replace_pyof_version(module_fullname, version):
...
+                                   (e.g.: pyof.v0x01.common.header)
             version (str): The version to be 'inserted' on the module fullname.
 
         Returns:
-            None (None): if the requested version is the same as the one of the
-                module_fullname or if the module_fullname is not a 'OF version'
-                specific module.
-            new_module_fullname (str): The new module fullname, with the
-                replaced version, on the format "pyof.v0x01.common.header".
+            str  : module fullname
```
Comment (diraol) [inline pyof/foundation/base.py]: Extra spaces

File: pyof/v0x01/asynchronous/error_msg.py [hunk 11]
```diff
@@ -185,7 +184,7 @@ def pack(self, value=None):
         After that, :attr:`data`'s value is restored.
 
         Returns:
-            bytes: The binary representation.
+            bytes : The binary representation.
```
Comment (diraol) [inline pyof/v0x01/asynchronous/error_msg.py]: Extra space

File: pyof/v0x01/common/phy_port.py [hunk 12]
```diff
@@ -133,13 +133,14 @@ class PhyPort(GenericStruct):
 
     The port_no field is a value the datapath associates with a physical port.
     The hw_addr field typically is the MAC address for the port;
-    :data:`.OFP_ETH_ALEN` is 6. The name field is a
+    :data:`OFP_ETH_ALEN` is 6. The name field is a
```
Comment (diraol) [inline pyof/v0x01/common/phy_port.py]: I think that the line break here is wrong...

File: pyof/v0x04/common/port.py [hunk 13]
```diff
@@ -50,8 +50,10 @@ class PortNo(IntEnum):
 class PortConfig(GenericBitMask):
     """Flags to indicate behavior of the physical port.
 
-    These flags are used in :class:`~Port` to describe the current
-    configuration. They are used in the :class:`~PortMod` message to configure
+    These flags are used in :class:`Port` to describe the current
+    configuration. They are used in the
+    :class:`~pyof.v0x04.controller2switch.port_mod.PortMod`
+    message to configure
```
Comment (diraol) [inline pyof/v0x04/common/port.py]: wrong line break here.

File: pyof/v0x04/common/table_feature.py [hunk 14]
```diff
@@ -293,7 +300,8 @@ def __init__(self, table_id=Table.OFPTT_ALL, name="",
             config(int):         Field reseved for future use.
```
Comment (diraol) [inline pyof/v0x04/common/table_feature.py]: some extra spaces here?

File: pyof/v0x04/common/table_feature.py [hunk 15]
```diff
@@ -293,7 +300,8 @@ def __init__(self, table_id=Table.OFPTT_ALL, name="",
...
             max_entries(int):    Describe the maximum number of flow entries
                                  that can be inserted into that table.
-            properties(ListOfProperty): List of Property intances.
+            properties(~pyof.v0x04.controller2switch.common.Property):
+                                 List of Property intances.
```
Comment (diraol) [inline pyof/v0x04/common/table_feature.py]: wrong indentation?

File: pyof/v0x04/controller2switch/common.py [hunk 16]
```diff
@@ -583,7 +606,8 @@ def __init__(self, table_id=Table.OFPTT_ALL, name="",
             config(int):         Field reseved for future use.
```
Comment (diraol) [inline pyof/v0x04/controller2switch/common.py]: extra spaces here?

File: pyof/v0x04/controller2switch/common.py [hunk 17]
```diff
@@ -583,7 +606,8 @@ def __init__(self, table_id=Table.OFPTT_ALL, name="",
...
             max_entries(int):    Describe the maximum number of flow entries
                                  that can be inserted into that table.
-            properties(ListOfProperty): List of Property intances.
+            properties(~pyof.v0x04.controller2switch.common.ListOfProperty):
+                                 List of Property intances.
```
Comment (diraol) [inline pyof/v0x04/controller2switch/common.py]: extra indentation here?

File: pyof/v0x01/common/phy_port.py (hunk 12 above)
Comment (macartur) [inline pyof/v0x01/common/phy_port.py]: This was fixed by other commit.

**Comment by diraol** [comment]
This PR was merged on a side branch and will me merged here soon.

**Comment by beraldoleal** [comment]
@diraol If there is a side branch with a solution ok, but lets keep the issue open until the merge be effective on this master.

**Pull request closed without merging**